### PR TITLE
[UXE-4871] feat: change events name and parse error message correctly in the ser…

### PR DESCRIPTION
**Pull request by aloisio-m-bastian**
…vices

## Pull Request
Fix network lists events name and fix error message handler in services
### Does this PR introduce breaking changes?
- [ ] No
- [ ] Yes 

### Does it have a link on Figma?

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by lucasmendes21 — APPROVED** [review]

**Review by lucasmendes21 — APPROVED** [review]

File: src/services/network-lists-services/create-network-list-service.js [hunk 1]
```diff
@@ -21,6 +21,33 @@ const adapt = (payload) => {
...
+        break
+      }
+    } else {
+      apiError = body[keyError]
+      break
+    }
+  }
+
+  return apiError
+}
```
Comment (HerbertJulio) [inline src/services/network-lists-services/create-network-list-service.js:49]:
```suggestion
/**
 * Extracts the first error message from an API response body.
 *
 * @param {Object} body - The response body.
 * @returns {string} The first error message found, or an empty string if none is found.
 */
const extractApiError = (body) => {
  for (const key in body) {
    const error = body[key];
    
    if (Array.isArray(error) && typeof error[0] === 'string') {
      return error[0];
    }
    
    if (Array.isArray(error) && error[0] && typeof error[0] === 'object' && typeof error[0].message === 'string') {
      return error[0].message;
    }
    
    if (typeof error === 'string') {
      return error;
    }
  }

  return '';
};
```

Comment (HerbertJulio) [inline src/services/network-lists-services/create-network-list-service.js:49]: Acho que podemos  simplificada para melhorar a legibilidade e manutenção o que acha?
```suggestion
const extractApiError = (body) => {
  for (const keyError of Object.keys(body)) {
    const errorValue = Array.isArray(body[keyError]) ? body[keyError][0] : body[keyError];
    if (typeof errorValue === 'string') return errorValue;
    if (typeof errorValue === 'object' && errorValue.message) return errorValue.message[0];
  }
  return '';
};
```

File: src/views/NetworkLists/EditView.vue [hunk 2]
```diff
@@ -74,15 +74,15 @@
...
+        productName: 'Network List'
       })
       .track()
   }
   const handleTrackFailEdit = (error) => {
     const { fieldName, message } = handleTrackerError(error)
     tracker.product
       .failedToEdit({
-        productName: 'Network Lists',
+        productName: 'Network List',
```
[the remark this answers is not in the document]
Comment (aloisio-m-bastian) [reply src/views/NetworkLists/EditView.vue:85]: o nome dos eventos está em singular 
image

 caso aqui estivessem no plural, seria enviado como "Created Network Lists", "Edit Network Lists e assim por diante"

File: src/services/network-lists-services/create-network-list-service.js [hunk 3]
```diff
@@ -21,6 +21,19 @@ const adapt = (payload) => {
   }
 }
 
+/**
+ * @param {Object} body - The response body.
+ * @returns {string} The result message based on the status code.
+ */
+const extractApiError = (body) => {
```
Comment (vinigfer) [inline src/services/network-lists-services/create-network-list-service.js:28]: Can we add a test for this?

File: src/services/network-lists-services/edit-network-list-service.js [hunk 4]
```diff
@@ -27,6 +27,19 @@ const adapt = (payload) => {
   }
 }
 
+/**
+ * @param {Object} body - The response body.
+ * @returns {string} The result message based on the status code.
+ */
+const extractApiError = (body) => {
+  for (const keyError of Object.keys(body)) {
```
Comment (vinigfer) [inline src/services/network-lists-services/edit-network-list-service.js:35]: And also this: is it possible to add a test?

File: src/services/network-lists-services/create-network-list-service.js (hunk 3 above)
Comment (aloisio-m-bastian) [reply src/services/network-lists-services/create-network-list-service.js:28]: done

**Review by lucasmendes21 — APPROVED** [review]

**Review by HerbertJulio — DISMISSED** [review]

**Review by peterpaulo-azion — APPROVED** [review]

**Review by vinigfer — APPROVED** [review]

**Review by HerbertJulio — APPROVED** [review]

**Review by vinigfer — APPROVED** [review]

**Pull request merged**